### ISSUE TITLE
Stream seems to provide too many bytes some times

### ISSUE THREAD

**Issue by joehoyle** [humanmade/S3-Uploads#201]
I've quite often seen these kind of errors when doing WP imports etc.

```
Warning: getimagesize(): corrupt JPEG data: 6203 extraneous bytes before marker in /srv/www/webroot/wordpress/wp-admin/includes/image.php on line 374
```

**Comment by jeremyfelt** [comment]
This has been driving me crazy for a bit now. We're seeing it with more and more images uploaded at WSU and I haven't found a common theme yet.

I do know that of the 4 uses of `wp_read_image_metadata` in core, only one of them isn't suppressing errors with `@wp_read_image_metadata()`. I kind of want to just propose that we hide this one too.

**Comment by rmccue** [comment]
I'd be OK with suppressing it if core is. I wonder if potentially core should be the one suppressing it instead though.

(I imagine these broken images are being passed through some broken image processing software which is giving them invalid data in the JPEG header.)

cc @joemcgill who might have thoughts on this

**Comment by jeremyfelt** [comment]
Joe from 4 years ago may have been on to something too :) https://core.trac.wordpress.org/ticket/28077#comment:3

**Comment by joehoyle** [comment]
I spend frankly way too long looking into this issue, and ended up opening a PHP core bug at https://bugs.php.net/bug.php?id=75708. If anyone wants to reproduce and add a code to the bug that would be great!

**Comment by joemcgill** [comment]
Just refreshed the patch which should fix this issue for WP – https://core.trac.wordpress.org/ticket/42480

Can someone confirm this addresses the issue?

**Comment by joehoyle** [comment]
@joemcgill so the warnings are a pain - however this isn't just an issue of the warnings. Per the above PHP bug, it can't actually get the metadata correctly when it's over a stream. I think it's probably worth the core patch I guess, but I see the "actual" issue here being the PHP bug.

**Comment by joemcgill** [comment]
Agreed that it makes sense to get PHP to address the real issue. Even so, if suppressing these warnings in Core helps and the patch addresses that issue, I'm happy to land it.
